src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -295,6 +294,24 @@ private async Task<Dictionary<OperationType, OpenApiOperation>> GetOperationsAsy
             {
                 await endpointOperationTransformer.TransformAsync(operation, operationContext, cancellationToken);
             }
+
+            var operationType = description.GetOperationType();
```

Comment:
This logic looks great overall! However, I believe we want to apply this before operation transformers are applied so that transformers registered by the user get the correct view.

I'm reviewing this on mobile so if this is already the case and I missed it let me know.

Also, we might want to add a test case to verify that transformers see the correct data.